### ISSUE TITLE
"unexpected end of stream on Connection" exception when trying to wrtie data in influxdb

### ISSUE THREAD

**Issue by leonard-IMBERT** [influxdb/influxdb-java#122]
Hi, i have an application wich need to write point in an influxdb. Here i recreate the error in a sbt project with a scala REPL : 
```
scala> import org.influxdb._
import org.influxdb._

scala> import org.influxdb.dto._
import org.influxdb.dto._

scala> val connection = InfluxDBFactory.connect("http://127.0.0.1:8086", "user", "password")
connection: org.influxdb.InfluxDB = org.influxdb.impl.InfluxDBImpl@7316c8b

scala> val point = Point.measurement("test").field("value", "ok")
point: org.influxdb.dto.Point.Builder = org.influxdb.dto.Point$Builder@2a8945dd

scala> connection.write("test_database", "all", point.build())
retrofit.RetrofitError: unexpected end of stream on Connection{127.0.0.1:8086, proxy=DIRECT hostAddress=127.0.0.1 cipherSuite=none protocol=http/1.1} (recycle count=0)
  at retrofit.RetrofitError.networkError(RetrofitError.java:27)
  at retrofit.RestAdapter$RestHandler.invokeRequest(RestAdapter.java:395)
  at retrofit.RestAdapter$RestHandler.invoke(RestAdapter.java:240)
  at org.influxdb.impl.$Proxy4.writePoints(Unknown Source)
  at org.influxdb.impl.InfluxDBImpl.write(InfluxDBImpl.java:156)
  at org.influxdb.impl.InfluxDBImpl.write(InfluxDBImpl.java:146)
  ... 43 elided
Caused by: java.io.IOException: unexpected end of stream on Connection{127.0.0.1:8086, proxy=DIRECT hostAddress=127.0.0.1 cipherSuite=none protocol=http/1.1} (recycle count=0)
  at com.squareup.okhttp.internal.http.HttpConnection.readResponse(HttpConnection.java:211)
  at com.squareup.okhttp.internal.http.HttpTransport.readResponseHeaders(HttpTransport.java:80)
  at com.squareup.okhttp.internal.http.HttpEngine.readNetworkResponse(HttpEngine.java:917)
  at com.squareup.okhttp.internal.http.HttpEngine.access$300(HttpEngine.java:95)
  at com.squareup.okhttp.internal.http.HttpEngine$NetworkInterceptorChain.proceed(HttpEngine.java:902)
  at com.squareup.okhttp.internal.http.HttpEngine.readResponse(HttpEngine.java:760)
  at com.squareup.okhttp.Call.getResponse(Call.java:274)
  at com.squareup.okhttp.Call$ApplicationInterceptorChain.proceed(Call.java:230)
  at com.squareup.okhttp.Call.getResponseWithInterceptorChain(Call.java:201)
  at com.squareup.okhttp.Call.execute(Call.java:81)
  at retrofit.client.OkClient.execute(OkClient.java:53)
  at retrofit.RestAdapter$RestHandler.invokeRequest(RestAdapter.java:326)
  ... 47 more
Caused by: java.io.EOFException: \n not found: size=0 content=...
  at okio.RealBufferedSource.readUtf8LineStrict(RealBufferedSource.java:200)
  at com.squareup.okhttp.internal.http.HttpConnection.readResponse(HttpConnection.java:191)
  ... 58 more

```
I use this version `"org.influxdb" % "influxdb-java" % "2.0"` of influxdb-java
Any idea why this happen ?

**Comment by mnuessler** [comment]
I was able to reproduce this error in both Scala and Java. Seems that the client is somehow sending an invalid request that InfluxDB cannot handle and causes it to panic with a go stacktrace. Then the client fails to parse the response because there is none. Was testing with InfluxDB version 0.9.4.1.

**Comment by mnuessler** [comment]
But it works if you write the point using a BatchPoint object:
```java
InfluxDB connection = InfluxDBFactory.connect("http://127.0.0.1:8086", "user", "password");
BatchPoints batchPoints = BatchPoints.database("test_database").build();
Point point = Point.measurement("test").field("value", "ok").build();
connection.write(batchPoints.point(point));
```

**Comment by mnuessler** [comment]
I think I found the problem. When calling the `write` method you are passing `all` as retention policy. This retention policy probably does not exist for that database, does it? Try to specify "default" (or the name of any other existing retention policy) when writing:
```java
connection.write("test_database", "default", point.build())
```

The error can also be reproduced using curl. It seems to be a bug in InfluxDB itself. If a retention policy does not exist, it should not panic.

This works:

    $ curl -i -XPOST 'http://localhost:8086/write?db=test_database' --data-binary "test value=\"ok\" 1448195072647000000"
    HTTP/1.1 204 No Content
    Request-Id: 7eb573cb-9125-11e5-800f-000000000000
    X-Influxdb-Version: 0.9.4.1
    Date: Sun, 22 Nov 2015 14:29:53 GMT

This works too:

    $ curl -i -XPOST 'http://localhost:8086/write?db=test_database&rp=default' --data-binary "test value=\"ok\" 1448195072647000000"
    HTTP/1.1 204 No Content
    Request-Id: 5de24a21-9125-11e5-800e-000000000000
    X-Influxdb-Version: 0.9.4.1
    Date: Sun, 22 Nov 2015 14:28:57 GMT

But with retention policy 'all' it fails:

    $ curl -i -XPOST 'http://localhost:8086/write?db=test_database&rp=all' --data-binary "test value=\"ok\" 1448195072647000000"
    curl: (52) Empty reply from server

**Comment by mnuessler** [comment]
By the way, this issue has already been fixed in InfluxDB version 0.9.5. It does not panic anymore, but returns an error response if the retention policy does not exist:

    $ curl -i -XPOST 'http://localhost:8086/write?db=test_database&rp=all' --data-binary "test value=\"ok\" 1448195072647000000"
    HTTP/1.1 500 Internal Server Error
    Request-Id: f2aa2fe5-9126-11e5-8002-000000000000
    X-Influxdb-Version: 0.9.5
    Date: Sun, 22 Nov 2015 14:40:17 GMT
    Content-Length: 32
    Content-Type: text/plain; charset=utf-8

    retention policy not found: all

**Comment by leonard-IMBERT** [comment]
Thanks a lot for your response,
I actually mixed up the consistency and the retention policie. Glad to see the new version of influxdb does not panic anymore :D
